### DISCUSSION TITLE
decoding using Fastspeech2 + conformer + HiFi vocoder jointly trained and use_teacher_forcing parameter

### DISCUSSION THREAD

**Discussion by BassamMattar** [espnet/espnet#4948] [Q&A] [answered]
I want to run the decoding stage in TTS using Fastspeech2 + conformer + HiFi-GAN vocoder.
I ran it with the "use_teacher_forcing"  parameter set to true
I found an error in inference:
TypeError: inference() got an unexpected keyword argument 'speech'
by tracing the code I found that the speech keyword is generated when use_teacher_forcing is set to true
so I set it to false and the decoding stage was completed successfully.
I wondered what is the functionality of use_teacher_forcing and how it affects the decoding ? as it is mentioned in the Readme of TTS task "The procedure is almost the same as FastSpeech but we **MUST** use teacher forcing in decoding."

**Comment by kan-bayashi** [comment, marked as answer]
Close via #4957
